### PR TITLE
Upgrade @apollo/client version to latest release candidate

### LINKED ISSUE #781: Upgrade to latest Apollo beta
### Description

Apollo beta released ~`vbeta.44`~ `vbeta.48` which has a breaking change to `cache.modify` that we are using. We should make sure to  make the appropriate changes to conform with the latest  API

### Acceptance Criteria

- Change all `cache.modify` calls to conform with the API as per [the latest version](https://github.com/apollographql/apollo-client/pull/6178)

- Confirm that all `cache.XXX` calls are stable

### PR DESCRIPTION
**Pull request by alexmylonas**
## Background

We are currently using a beta version of `@apollo/client`(specifically `v3.0.0-beta.41`). 

Apollo recently released `v3.0.0-rc.0` and the sooner we upgrade version the less conflicts we 'll have in code.

One of the core changes that affects our implementation is explained here: [6178](https://github.com/apollographql/apollo-client/pull/6178) & [6350](https://github.com/apollographql/apollo-client/pull/6350)

Closes #781 

## Changes

- Updated `package.json` and `packege-lock.json`
- Updated all `cache.modify` instances based on it's new syntax

## Testing

- Locally
- Deployed env

### CODE REVIEW

**Comment by 3nvi** [comment]
@alexmylonas due to too many upcoming changes, I'd like this merged after #1088 is merged. Is that ok with you?

**Comment by alexmylonas** [comment]
> @alexmylonas due to too many upcoming changes, I'd like this merged after #1088 is merged. Is that ok with you?

Sure thing, not a high prio anyways.

File: package.json [hunk 1]
```diff
@@ -88,7 +88,7 @@
     "webpack-manifest-plugin": "^2.2.0"
   },
   "dependencies": {
-    "@apollo/client": "3.0.0-beta.41",
+    "@apollo/client": "3.0.0-rc.0",
```
Comment (3nvi) [inline package.json:91]: let's bump to latest rc (has more bug fixes)

Comment (alexmylonas) [reply package.json]: Updated to `v3.0.0-rc.12`

**Comment by alexmylonas** [comment]
@3nvi Upgrade to latest apollo client (`v3.0.0-rc.12`) and change new introductions of cache.modify.

I am merging this!

**Pull request merged**